stl/inc/chrono

```diff
@@ -5171,6 +5176,180 @@ namespace chrono {
+
+        constexpr void _On_precision(int _Prec) {
+            _Specs._Precision = _Prec;
+        }
+
+        constexpr void _On_conversion_spec(_CharT _Modifier, _CharT _Type) {
+            // NOTE: same performance note from _Basic_format_specs also applies here
+            const char _Char_mod  = static_cast<char>(_Modifier);
+            const char _Char_type = static_cast<char>(_Type);
+            if (_Char_mod != '\0' && _Char_mod != 'E' && _Char_mod != 'O') {
+                _THROW(format_error("Invalid modifier specification."));
+            }
+
+            if (_Type < 0 || _Type > (numeric_limits<signed char>::max)()) {
+                _THROW(format_error("Invalid type specification."));
+            }
+
+            _Chrono_specs<_CharT> _Conv_spec{._Modifier = _Char_mod, ._Type = _Char_type};
+            _Specs._Chrono_specs_list.push_back(_Conv_spec);
+        }
+
+        constexpr void _On_lit_char(_CharT _Lit_ch) {
+            _Chrono_specs<_CharT> _Lit_char_spec{._Lit_char = _Lit_ch};
+            _Specs._Chrono_specs_list.push_back(_Lit_char_spec);
+        }
+
+    protected:
+        _Chrono_format_specs<_CharT>& _Specs;
+    };
+
+    template <class _CharT, _Chrono_parse_spec_callbacks<_CharT> _Callbacks_type>
+    _NODISCARD constexpr const _CharT* _Parse_conversion_specs(
+        const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+        ++_Begin; // move past %
+        if (_Begin == _End || *_Begin == '}') {
+            _THROW(format_error("Invalid format string."));
+        }
+
+        _CharT _Mod = '\0';
+        _CharT _Ch  = *_Begin;
+
+        if (_Ch == 'E' || _Ch == 'O') { // includes modifier
+            _Mod = _Ch;
+            ++_Begin;
+            if (_Begin == _End || *_Begin == '}') {
+                _THROW(format_error("Invalid format string - missing type after modifier."));
+            }
+        }
+
+        _CharT _Type = *_Begin;
+        _Callbacks._On_conversion_spec(_Mod, _Type);
+
+        return ++_Begin;
+    }
+
+    template <class _CharT, _Chrono_parse_spec_callbacks<_CharT> _Callbacks_type>
+    _NODISCARD constexpr const _CharT* _Parse_chrono_format_specs(
+        const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+        if (_Begin == _End || *_Begin == '}') {
+            return _Begin;
+        }
+
+        _Begin = _Parse_align(_Begin, _End, _Callbacks);
+        if (_Begin == _End) {
+            return _Begin;
+        }
+
+        _Begin = _Parse_width(_Begin, _End, _Callbacks);
+        if (_Begin == _End) {
+            return _Begin;
+        }
+
+        if (*_Begin == '.') {
+            _Begin = _Parse_precision(_Begin, _End, _Callbacks);
+            if (_Begin == _End) {
+                return _Begin;
+            }
+        }
+
+        // chrono-spec
+        while (_Begin != _End && *_Begin != '}') {
+            if (*_Begin == '%') { // conversion-spec
+                if (_Begin + 1 == _End) {

```

Comment:
The whole `_Begin + 1` is only needed because you increment `_Begin` as the first instruction in `_Parse_conversion_specs`

I believe it would considerable simplify the code if we would pull that out

Reply:
I was trying to echo what other functions like `_Parse_precision` in `<format>` did.  It's the only example that's similar to this case so I'm not hugely opposed to changing this as suggested when we implement the next bit of "chronat".